packages/components/data-table/src/use-manual-column-resizing-reducer.ts

```diff
@@ -1,20 +1,74 @@
-import { useReducer, useDebugValue } from 'react';
+import { useReducer, useDebugValue, MutableRefObject, Dispatch } from 'react';
+import type { TColumn } from './data-table';
+
+type TTableRef = MutableRefObject<HTMLTableElement | undefined>;
+
+type TState = {
+  tableRef?: TTableRef;
+  sizes?: TRenderedColumnMeasurements[];
+  initialColWidth?: number;
+  initialMousePosition?: number;
+  columnBeingResized?: number;
+  hasBeenResized?: boolean;
+};
+
+export type TRenderedColumnMeasurements = {
+  key: string | null;
+  width: number;
+};
+
+type TPayload = {
+  initialColWidth: undefined;
+  initialMousePosition: undefined;
+  columnBeingResized: undefined;
+  hasBeenResized: boolean;
+  sizes: undefined;
+  tableRef: TTableRef;
+};
+
+type TStartResizingPayload = {
+  initialColWidth: number;
+  initialMousePosition: number;
+  columnBeingResized: number;
+} & TState;
+
+type TFinishResizingPayload = {
+  initialColWidth: undefined;
+  initialMousePosition: undefined;
+  columnBeingResized: undefined;
+  hasBeenResized: true;
+} & TState;
+
+type TRegisterColumnMeasurements = {
+  sizes: TRenderedColumnMeasurements[];
+} & TState;
+
+type TAction =
+  | { type: 'reset'; payload?: TPayload }
+  | { type: 'registerColumnMeasurements'; payload: TRegisterColumnMeasurements }
+  | { type: 'startResizing'; payload: TStartResizingPayload }
+  | { type: 'finishResizing'; payload?: TFinishResizingPayload }
+  | { type: undefined };
```

Comment:
Hmm this looks weird. Why do we need this?

Reply:
We need it to properly type the action parameter in the reducer function.

Reply:
I'm not following. Having an action like `type: undefined` does not make much sense.

What is the error you get without this?

Reply:
I get this
![image](https://user-images.githubusercontent.com/6726467/141980908-582a4605-8b62-4f81-b7b6-4befd6a38c47.png)

Reply:
Ah yes. We don't need to throw an error. For the `default` case we should simply do `return state;`.

Reply:
Okay, that makes sense. 
Updated.